src/mg/PAMI/Client/Impl/ClientImpl.php

```diff
@@ -443,11 +439,14 @@ public function close()
 	 */
 	public function __construct(array $options)
 	{
-        if (isset($options['log4php.properties'])) {
-            \Logger::configure($options['log4php.properties']);
-        }
-        $this->_logger = \Logger::getLogger('Pami.ClientImpl');
-	    $this->_host = $options['host'];
+		$this->setLogger(
```

Comment:
Excuse me? Could you please explain why? At least in the PHP world, calling setters from the constructor is perfectly acceptable.

One should not do actual *work* from the constructor, but validating parameters, instantiating objects (smelly, but OK) and setting properties is not a problem.